### PR TITLE
feat(addon-editor): use `https` by default for links in editor

### LINKED ISSUE #4044: 🚀 - Support https links in tui-editor
### Which @taiga-ui/* package(s) are relevant/related to the feature request?

addon-editor

### Description

image

UIKIT-4801

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #4044

image

## What is the new behavior?

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/addon-editor/tokens/editor-options.ts [hunk 1]
```diff
@@ -3,18 +3,21 @@ import {
...
     readonly colors: ReadonlyMap<string, string>;
     readonly fontOptions: typeof tuiDefaultFontOptionsHandler;
+    readonly linkOptions?: typeof tuiDefaultLinkOptionsHandler;
 }
 
 export const TUI_EDITOR_DEFAULT_OPTIONS: TuiEditorOptions = {
     colors: defaultEditorColors,
     blankColor: EDITOR_BLANK_COLOR,
     fontOptions: tuiDefaultFontOptionsHandler,
+    linkOptions: tuiDefaultLinkOptionsHandler,
```
Comment (waterplea) [inline projects/addon-editor/tokens/editor-options.ts:20]: Why is it a function?

File: projects/addon-editor/constants/default-link-options-handler.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export const TUI_EDITOR_LINK_HASH_PREFIX = `#` as const;
+export const TUI_EDITOR_LINK_HTTP_PREFIX = `http://` as const;
+export const TUI_EDITOR_LINK_HTTPS_PREFIX = `https://` as const;
+
+export type TuiEditorLinkProtocol =
+    | typeof TUI_EDITOR_LINK_HTTP_PREFIX
+    | typeof TUI_EDITOR_LINK_HTTPS_PREFIX;
+
+export type TuiEditorLinkPrefix =
```
Comment (waterplea) [inline projects/addon-editor/constants/default-link-options-handler.ts:9]: `= TuiEditorLinkProtocol | typeof TUI_EDITOR_LINK_HASH_PREFIX`

File: projects/addon-editor/components/edit-link/edit-link.component.ts [hunk 3]
```diff
@@ -48,21 +53,27 @@ export class TuiEditLinkComponent {
         @Inject(TUI_EDITOR_LINK_TEXTS)
         readonly texts$: TuiInjectionTokenType<typeof TUI_EDITOR_LINK_TEXTS>,
         @Inject(TuiTiptapEditorService) private readonly editor: AbstractTuiEditor,
+        @Inject(TUI_EDITOR_OPTIONS)
+        private readonly options: TuiEditorOptions,
     ) {}
 
+    get defaultProtocol(): TuiEditorLinkProtocol {
+        return this.options?.linkOptions?.()?.protocol ?? TUI_EDITOR_LINK_HTTP_PREFIX;
```
Comment (waterplea) [inline projects/addon-editor/components/edit-link/edit-link.component.ts:61]: `this.options` cannot be falsy. And shouldn't default fallback be https?

File: projects/addon-editor/tokens/editor-options.ts (hunk 1 above)
Comment (splincode) [reply projects/addon-editor/tokens/editor-options.ts]: fixed

File: projects/addon-editor/constants/default-link-options-handler.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-editor/constants/default-link-options-handler.ts:9]: fixed

File: projects/addon-editor/components/edit-link/edit-link.component.ts (hunk 3 above)
Comment (splincode) [reply projects/addon-editor/components/edit-link/edit-link.component.ts]: fixed

File: projects/addon-editor/components/edit-link/edit-link.component.ts [hunk 4]
```diff
@@ -48,21 +53,27 @@ export class TuiEditLinkComponent {
         @Inject(TUI_EDITOR_LINK_TEXTS)
         readonly texts$: TuiInjectionTokenType<typeof TUI_EDITOR_LINK_TEXTS>,
         @Inject(TuiTiptapEditorService) private readonly editor: AbstractTuiEditor,
+        @Inject(TUI_EDITOR_OPTIONS)
+        private readonly options: TuiEditorOptions,
     ) {}
 
+    get defaultHttpProtocol(): TuiEditorLinkProtocol {
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-editor/components/edit-link/edit-link.component.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**